src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -374,12 +374,10 @@ private int WritePosixName(Span<byte> buffer)
 
             if (_name.Length > FieldLengths.Name)
             {
-                int prefixBytesLength = Math.Min(_name.Length - FieldLengths.Name, FieldLengths.Name);
-                Span<byte> remaining = prefixBytesLength <= 256 ?
-                    stackalloc byte[prefixBytesLength] :
-                    new byte[prefixBytesLength];
-
-                int encoded = Encoding.ASCII.GetBytes(_name.AsSpan(FieldLengths.Name), remaining);
+                int prefixBytesLength = Math.Min(_name.Length - FieldLengths.Name, FieldLengths.Prefix);
+                Debug.Assert(prefixBytesLength <= 256);
```

Comment:
@Jozkee this should be 155. The above Math.Min calculation should never give us a number larger than that, which is the max length we can fit in the prefix field.
```suggestion
                Debug.Assert(prefixBytesLength <= 155);
```

Reply:
If you fix this, we can execute runtime-extra-platforms too.

Reply:
this was meant to honor the former condition:
```cs
Span<byte> remaining = prefixBytesLength <= 256 ?
    stackalloc byte[prefixBytesLength] :
    new byte[prefixBytesLength];
```
But I can change it, we can also remove it as it is obvious that it will always be <= 155, unless FieldLengths.Prefix constant changes which seems unlikely.

Reply:
It won't change. The prefix field was defined in the IEEE Std 1003.1-1988 ("POSIX.1") spec (see the ["POSIX ustar Archives" section](https://www.freebsd.org/cgi/man.cgi?tar(5))), and was then inherited by Pax, which is supposed to be backwards compatible with Ustar.